### PR TITLE
[NEWDEV] Prevent Path from being an empty or NULL string

### PR DESCRIPTION
**Pull request by zecarlos1957**
## Purpose

Prevent Path from being an empty or NULL string

### CODE REVIEW

File: dll/win32/newdev/newdev.c [hunk 1]
```diff
@@ -353,7 +353,10 @@ SearchDriverRecursive(
     BOOL retval = FALSE;
     HANDLE hFindFile = INVALID_HANDLE_VALUE;
 
-    wcscpy(DirPath, Path);
+    if (Path && Path[0])
+        wcscpy(DirPath, Path);
+    else
+        return retval;
```
Comment (SergeGautherie) [inline dll/win32/newdev/newdev.c:359]:
```suggestion
    if (!Path || !Path[0])
        return retval;

    wcscpy(DirPath, Path);
```

**Comment by JoachimHenze** [comment]
How can we trigger that crash with the wcscpy in practice? What were your reproduction steps?

File: dll/win32/newdev/newdev.c (hunk 1 above)
Comment (zecarlos1957) [reply dll/win32/newdev/newdev.c]: Of course, mutch beter.
The suggested committee added changes to the header???

**Comment by zecarlos1957** [comment]
@JoachimHenze

**Comment by zecarlos1957** [comment]
@JoachimHenze   In fact, I've never seen a crash here. But I was warned by someone about this danger.

File: dll/win32/newdev/newdev.c (hunk 1 above)
Comment (SergeGautherie) [reply dll/win32/newdev/newdev.c]: > The suggested commit added changes to the header???

It seems so, for unknown reason.

**Comment by HBelusca** [comment]
Not going to approve unless someone can tell me who is calling SearchDriverRecursive with a NULL Path pointer. The function clearly states from its annotation that it doesn't accept a NULL pointer there, already, so this current hack just hides a bug that is elsewhere.

**Pull request closed without merging**